packages/investment-pool/contracts/InvestmentPool.sol

```diff
@@ -784,37 +818,44 @@ contract InvestmentPool is
         _gelatoTransfer(fee, feeToken);
     }
 
-    
-
     // TODO: Ensure that this wouldn't clash with our logic
     // Introduce limits and checks to prevent gelato from taking too much
     function _gelatoTransfer(uint256 _amount, address _paymentToken) internal {
         if (_paymentToken == 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE) {
             // If ETH address
             (bool success, ) = gelato.call{value: _amount}("");
-            require(success, "[IP] _gelatoTransfer: ETH transfer failed");
+            if (!success) revert InvestmentPool__GelatoEthTransferFailed();
         } else {
             // Else it is ERC20 token
             SafeERC20.safeTransfer(IERC20(_paymentToken), gelato, _amount);
         }
     }
 
-    function _calculatePortion(uint256 _amountTotal, uint256 _portion) internal returns (uint256) {
-        
-        return _amountTotal * _portion / PERCENTAGE_DIVIDER;
+    function _calculatePortion(uint256 _amountTotal, uint256 _portion)
+        internal
+        pure
+        returns (uint256)
+    {
+        return (_amountTotal * _portion) / PERCENTAGE_DIVIDER;
     }
 
     // NOTE: We don't need that?
-    function _payoutMissedInvestmentCompensations(uint256 _milestoneId) internal {
+    function _payoutMissedInvestmentCompensations(uint256 _milestoneId)
```

Comment:
It's probably time we get rid of this, along with `_calculatePortion(uint256 _amountTotal, uint256 _portion)`